### PR TITLE
Use .NET configuration to read --debug flag value

### PR DESCRIPTION
**Pull request by calebkiage**

### CODE REVIEW

**Comment by calebkiage** [comment]
Ready since #164 is approved. Generated code will need microsoft/kiota#2268

File: src/Microsoft.Graph.Cli.Core/Configuration/Extensions/CommandArgumentsExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        for (int i = 0; i < args.Length; ++i)
+        {
+            var curr = args[i];
+            argsSanitized.Add(curr);
+            if (!curr.StartsWith('-')) continue;
+            string? next = null;
+            if (i < args.Length - 1)
+            {
+                next = args[i + 1];
+            }
```
Comment (baywet) [inline src/Microsoft.Graph.Cli.Core/Configuration/Extensions/CommandArgumentsExtensions.cs:20]:
```suggestion
            if (!curr.StartsWith('-') || i > args.Length - 1) continue;
            var next = i == args.Length - 1 ? string.Empty : args[i + 1];
```

File: src/Microsoft.Graph.Cli.Core/Configuration/Extensions/CommandArgumentsExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+            if (curr.StartsWith('-') && (next?.StartsWith('-') == true || i == args.Length - 1))
```
Comment (baywet) [inline src/Microsoft.Graph.Cli.Core/Configuration/Extensions/CommandArgumentsExtensions.cs:22]:
```suggestion
            if (next.StartsWith('-') || i == args.Length - 1)
```

**Review by baywet — CHANGES_REQUESTED** [review]

**Review by baywet — APPROVED** [review]
even better with the last change!

**Comment by calebkiage** [comment]
Thanks for pointing out the optimization opportunity there. :-)

**Pull request merged**